napari/layers/shapes/shape_list.py

```diff
@@ -90,6 +112,48 @@ def z_indices(self):
         """list of int: z-index for each shape."""
         return [s.z_index for s in self.shapes]
 
+    @property
+    def slice_key(self):
+        """list: slice key for slicing n-dimensional shapes."""
+        return self._slice_key
+
+    @slice_key.setter
+    def slice_key(self, slice_key):
+        slice_key = list(slice_key)
+        if not np.all(self._slice_key == slice_key):
+            self._slice_key = slice_key
+            self._update_displayed()
+
+    def _update_displayed(self):
+        """Update the displayed data based on the slice key."""
+        slice_key = np.array([self.slice_key, self.slice_key])
```

Comment:
I don't understand the repetition here...

Reply:
it is to check against both the min and max value, i'll add a comment - maybe there's a better way to do this that we can implement in the future